code/WorkInProgress/Cael_Aislinn/Rust/fuel_compressor.dm

```diff
@@ -75,17 +203,24 @@ var/const/max_assembly_amount = 300
 		else
 			to_chat(usr, "<span class='warning'>\icon[src] there are no more compressed matter units in [src].</span>")
 
+	if ( href_list["purge_matter"] )
+		switch(alert("Do you want to purge the compressed matter in storage?","Please Confirm","Yes","No"))
+			if("Yes")
+				compressed_matter = 0
+			if("No")
```

Comment:
On another note: you will have to make sure the user is conscious and adjacent to the machine after this `alert` in case anything happened to them before they accepted.